### DISCUSSION TITLE
`install-info` not shipped on macOS Ventura

### DISCUSSION THREAD

**Discussion by ywwry66** [Homebrew/discussions#4306] [Everyday usage] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.0.4
ORIGIN: https://github.com/Homebrew/brew
HEAD: 4d0154fa7926e8a7b06e9b074dd45957e820407c
Last commit: 6 days ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 3f42a5df175e9f2b8646a85c923acf0bbf692b28
Core tap last commit: 6 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.oTVMzuaa43/org.xquartz:0
HOMEBREW_EDITOR: emacs
HOMEBREW_MAKE_JOBS: 10
HOMEBREW_NO_INSTALL_FROM_API: set
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 14.0.0 build 1400
Git: 2.39.2 => /opt/homebrew/bin/git
Curl: 7.86.0 => /usr/bin/curl
macOS: 13.2.1-arm64
CLT: 14.2.0.0.1.1668646533
Xcode: N/A
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

`install-info` is not shipped on macOS Ventura, which caused following functions not working with `brew install` and `brew uninstall` commands:
https://github.com/Homebrew/brew/blob/1447f0853bb2e9006e89d8db0289cfef9e7aa05b/Library/Homebrew/extend/pathname.rb#L346-L354
As a result, the `dir` file in `/opt/homebrew/share/info/` is not updating anymore.

Strictly speaking, this is not a `brew` issue. But since macOS is `brew`'s main platform, I was wondering if there is anything Homebrew can do to make this work again?

**Comment by ywwry66** [comment, marked as answer]
Is it ok just to replace `/usr/bin/install-info` with `install-info` in `pathname.rb`? This way any `install-info` in users' `PATH` will do the job. We can also regenerate the entire `dir` file in the `post_install` block of the `texinfo` formula, so that everything will work out of the box with `brew`ed `texinfo`. If this makes sense, I can submit RPs in related repos.

**Reply by carlocab** [reply]
Just using `install-info` won't work, since `$HOMEBREW_PREFIX/bin` is not in `PATH` when running inside `brew`. (`PATH` is filtered [here](https://github.com/Homebrew/brew/blob/8a5e1697a64ab8b694fc1d4fd6c8e948fe6950be/bin/brew#L151-L152).) You could probably do something like [this](https://github.com/Homebrew/brew/blob/8a5e1697a64ab8b694fc1d4fd6c8e948fe6950be/Library/Homebrew/formula_cellar_checks.rb#L300-L301), though. We'll review a PR.

**Reply by ywwry66** [reply]
Thanks for pointing the correct direction. I will take a look.
